### PR TITLE
Context Data and Intents consolidated update PR

### LINKED ISSUE #574: StartChat and SendChatMessage
### Use Case:

This proposal covers two connected use cases:
- extend the chat creation option parameters (today we only support a contact or a list of contacts)
- support the possibility to send messages in existing chats

#### StartChat

The proposal is to support a new context type (ChatInit) to initialize a chat creation with new optional parameters:
- a chat name
- an initial pre-formatted message (that would be edited prior to sending in the chat application)
- other application-specific configuration options (cf. below)

Example:
```ts
let chatInit: ChatInit = {
    type: "fdc3.chatInit",
    chatName: "Deal ABCD",
    members: {
        type: "fdc3.contactList",
        contacts: [
            {
                type: "fdc3.contact",
                name: "John Doe",
                id: {
                    "email": "john@sample.com"
                }
            }
        ]
    },
    chatRoomOptions: {
        "symphony": {
            visibility: "EXTERNAL-SINGLE",
            discoverable: true
        }
    },
    initMessage: "Hi John"
}
```

Which would translate into this type of interface:

```ts
// Intent<ParamType, ReturnType>
type StartChat = Intent<Contact | ContactList | ChatInit, ChatRoomRef>;

/**
 * ChatInit: new context to define the initialization parameters to create a new chat
 */
interface ChatInit {
    type: "fdc3.chat.init";
    chatName?: string;
    members: ContactList;
    chatRoomOptions?: { [appName: string]: ChatOptions };
    initMessage?: string | RichTextBlock;
}

type ChatOptions = Object;

/**
 * ChatOptions specific to Symphony
 */
interface SymphonyChatOptions extends ChatOptions {
    type: "symphony.chat.initOptions";
    visibility: "PUBLIC" | "PRIVATE" | "EXTERNAL-SINGLE" | "EXTERNAL-MULTI";
    discoverable?: boolean;
    memberPermissions?: {
        allowHistoryBrowsing?: boolean;
        allowMessageSend?: boolean;
        allowMessageCopy?: boolean;
        allowAddUser?: boolean;
    }
}

interface ChatRoomRef {
    type: "fdc3.chat.roomRef";
    appName: string; // name of the application that owns this chat - e.g. "symphony"
    id: {[idType:string]: string}; // id Object - similar structure 
    name?: string; // name of the chat at the time of its creation
}
```

Notes:
- the purpose of the dictionary structure for **chatRoomOptions** is to allow supporting multiple chat applications. If multiple options are passed, the FDC3 agent should ask the end user to choose which application to pick (e.g. Symphony or Teams)
- the optional init message could be either a plain string or a formatted structure (to be discussed in a separate proposal)
- we also propose to have the StartChat intent return a **ChatRoomRef** object that could be used in the **SendChatMessage** to send further messages in the same chat room (cf. separate proposal)

#### SendChatMessage

This new intent would be used to send a message into an existing chat (that may have previously been created through StartChat).

```ts
// example
let chatMessage: ChatMessage = {
    type: "fdc3.chat.message",
    chatRoom: {
        type: "fdc3.chat.roomRef",
        appName: "symphony",
        id: "KTHwU5nxJJDgTKaIshGvjI3///o3"
    },
    message: "Hello World"
}
```

**SendChatMessage** would take a **ChatMessage** as input and would return 
- either null if the message was not sent (e.g. the end-user changed her mind)
- or a **ChatRoomRef** context if the message was sucessfully sent. The reason for sending back a ChatRoomRef context is to give an application the opportunity to receive an updated version of the chat name (that may have been changed since its creation).

```ts
// Intent<ParamType, ReturnType>
type SendChatMessage = Intent<ChatMessage, ChatRoomRef | null>;

/**
 * ChatMessage: context passed to SendChatMessage
 * Send a message in an existing chat conversation
 */
interface ChatMessage {
    type: "fdc3.chat.message";
    chatRoom: ChatRoomRef;
    message: string | RichTextBlock;
}

/**
 * Context returned by StartChat and SendChatMessage
 */
interface ChatRoomRef {
    type: "fdc3.chat.roomRef";
    appName: string; // name of the application that owns this chat - e.g. "symphony"
    id: string;
    name?: string; // name of the chat
}
```

Note: the RichTextBlock type will be discussed in a separate thread

### LINKED ISSUE #587: Intent Proposal: ViewResearch
### Use Case
A Salesperson is using an IM system to discuss the performance of a set of particular stocks with a Buy Side Fund Manager and mentions that their analyst has published a well-regarded report on one of those stocks. The fund manager hasn’t seen the report and requests that the Salesperson sends it to them. The Salesperson needs to open the Research Management System in order to access the research and generate a link they can IM to the fund manager.

An Analyst is on a call with a Buy Side analyst who requests a copy of their latest model on a company under discussion. They have details of the latest research read by the analyst and uses the ticker for the company report to open the Research Management System page for all research on that stock.

#### ViewResearch
ViewResearch should be used when a user wants to see the latest research on a particular stock. ViewResearch would be called with the Instrument context.

### Additional Information
There may be some overlap between ViewResearch and ViewAnalysis in the sense that research is a form of analysis. However the purpose of ViewAnalysis seems to be broader that ViewResearch which is specifically about getting the latest published on a stock.

### LINKED ISSUE #588: Intent Proposal: ViewOrders
### Use Case
A SalesTrader is using the CRM to research candidates for a series of trade opportunities. Based on information such as current holdings, changes in position, research consumption, etc. a number of companies are found and for each of them the SalesTrader then wants to review detailed order history in the OMS to see if each organisation is likely to participate in the trades.

#### ViewOrders
ViewOrders should be used when a user wants to see the order history of an individual, an institution or of a particular stock. ViewResearch would be called with the Contact, Organization or Instrument contexts.

### LINKED ISSUE #589: Intent Proposal: ViewInteractions
### Use Case
A Salesperson is IMing with a Buy Side fund manager who mentions that they haven’t been receiving good services from the analyst covering the sector they specialise in. The Salesperson thinks this isn’t right as she knows the analyst was in a meeting with the fund manager’s firm and so they need to use the CRM to check whether the meeting they’re thinking of went ahead or not.

A Corporate Client Manager is preparing for a client meeting. They start in the BookBuilding system gathering data about a recent fund raise they led and then wants to flip to the CRM to see all the feedback from institutions that attended the related roadshow.

#### ViewInteractions
ViewInteractions should be used when a user wants to see the latest interactions (calls, meetings, conferences, roadshows) on a particular stock or with an individual or organisation. ViewInteraction would be called with the Contact, Organization or Instrument contexts.

### LINKED ISSUE #590: Intent Proposal: StartEmail
### Use Case
A Salesperson is IMing with a Buy Side fund manager who requests a follow-up set of information including research links and some detail on upcoming meetings. The Salesperson promises to send over an email with the relevant information and so wants to open a draft email to the fund manager to start the process off. One click takes the user to the CRM with the relevant individual selected as the recipient, the regular template for client emails opened and the research archive available to quickly add research links.

#### StartEmail
StartEmail should be used when a user wants to create an email to send to an individual or a list of individuals. StartEmail would be called with the Email context.

### Contexts

The  fdc3.Nothing Context can be provided to initiate the Start Email workflow with a blank email

#### Email
The Email Context allows the passing of all the relevant information to provide the user with a well-drafted email.

| Property    | Type    | Required | Example Value       |
|-------------|---------|----------|---------------------|
| `type`      | string  | Yes      | `'fdc3.email'` |
| `recipients`      | fdc3.Contact or fdc3.ContactList  | Yes       | { type: "fdc3.contact", name: "John Doe", id: { "email": "john@sample.com"}} |
| `subject` | string  | No       | `'The information you requested'`            |
| `textBody`    | string  | No       | `'Blah, blah, bah`         |
| `richTextBody`    | richtextblock  | No       | See https://github.com/finos/FDC3/issues/575        |

### Example

```
window.fdc3.raiseIntent('fdc3.StartEmail', {
    type: 'fdc3.email',
    recipients: {
        type: 'fdc3.contact',
        name: 'Jane Doe',
        id: {
            email: 'jane.doe@example.com'
        }
    },
    subject: 'The information you requested',
    textBody: 'Blah, blah, blah ...'
})
```

### Additional Information
This rounds out the StartCall and StartChat intents to cover all the main communication methods.

### LINKED ISSUE #600: Intent Proposal: View Profile
### Use Case
A Corporate Access administrator is researching a potential invite list for an upcoming IPO. To do this they look at data held in multiple systems such as their Order Management System, their Bookbuilding System, etc in order to determine which of the organisations who have previously participated in offerings may be likely to participate in the IPO they are prepping. Once they have determined which organisations to target, they will need to figure out which individuals at that organisation to approach. This information will be held in the CRM.

A salesperson is about to initiate a chat with a Buy Side fund manager on IM and just wants to check their details; job title, investment style, before they pick up the conversation. This information will be held in the CRM

#### ViewProfile
ViewProfile would allow a quick navigation to an individual's or a company’s 'basic information'. ViewProfile would be called with the Contact or Organization context.

### Additional Information
ViewProfile with the Contact context is essentially the same as the current ViewContact intent. This could be deprecated over time but, for the time being, they should be considered equivalent.

This issue was created after feedback on a proposal for ViewOrganization to make it more general https://github.com/finos/FDC3/issues/586

### LINKED ISSUE #645: Standardize recommended formats for 'primitive' Context fields
## Enhancement Request

Add recommended formats (SHOULD) for common field types to the Context specification, where those fields don't (or don't often or always) warrant a full embedded context type. Those field types should reference external standards where possible.

Note recommending a format for fields of these types doesn't preclude them *also* having a context type. However, if it is decided that they should *always* be full context types then that should instead be _required_ in the standard and this information distributed amongst each individual type.

### Use-case
Providing recommended formats will reduce or prevent competing conventions from being established in both standardized types and in-house types created by app developers - helping to ensure applications can interoperate through FDC3 more reliably. 

Further, conversion of a format or code is often trivial if the target format is specified and it can be validated by a schema. Whereas interpretation of diverse formats from a single field can be far more difficult as is validation via a schema.

### Examples
For example, the following types of data might have a recommended format:
- Time
  -  [ISO 8601-1:2019](https://www.iso.org/standard/70907.html) (already linked to in [References page](https://fdc3.finos.org/docs/next/references))
    - A timezone indicator should be included
    - E.g. 
      - Time in UTC: `2022-03-30T15:44:44Z`
      - Also time in UTC: `2022-03-30T15:44:44+00:00`
      - Same time in EDT: `2022-03-30T11:44:44-04:00`
    - Parsing in Javascript: `let aDate = new Date("2022-03-30T11:44:44-04:00")`
- Date
  - `YYYY-MM-DD` date format from [ISO 8601-1:2019](https://www.iso.org/standard/70907.html) (already linked to in [References page](https://fdc3.finos.org/docs/next/references))
    - E.g. `2022-03-30`
    - Parsing in Javascript: `let aDate = new Date("2022-03-30")`
- Country code
  - Alpha-2-code from [ISO 3166-1](https://www.iso.org/iso-3166-country-codes.html)  (already linked to in [References page](https://fdc3.finos.org/docs/next/references))
    - Note current fdc3.Country type uses *both* the Alpha-2 and Alpha-3 codes - picking one would simplify requirements
  - E.g. `GB` or `UA`
- Currency code 
  - Alphabetic code from [ISO 4217](https://www.iso.org/iso-4217-currency-codes.html) (needs adding to [References page](https://fdc3.finos.org/docs/next/references))
  - E.g. `GBP` or `UAH`

### LINKED ISSUE #646: Standardize a context type representing a range of time: TimeRange
## Enhancement Request (edited 6th May 2022)

Create a context type that can be used to represent a period of time.

### Use Case:
Any user interfaces that represent or visualise events or activity over time can be filtered or focused on a particular time period, e.g.:
- A pricing chart
- A trade blotter
- An record of client contact/activity in a CRM

A time period may be closed (i.e. start and end date are both known) or open (i.e. only the start date is known, an end date will be known in the future, e.g. when an order is filled).

Example application in a raised intent: 
> User may want to view pricing/trades/customer activity for a security over a particular time period, the time range might be specified as the context for the `ViewChart` intent OR it be embedded in another context (e.g. a Chart context).
  
Example application as a broadcast context:
> User filters a visualisation (e.g. a pricing chart) to show a particular period, the TimeRange is broadcast, other visualisations (e.g.  a heatmap of activity by instrument, or industry sector etc.) receive it and filter themselves to show data over the same range.

#### TimeRange
Represents a specific period of time.

##### Details

| Property    | Type    | Required | Example Value       |
|-------------|----------|----------|---------------------|
| `type`         | string    | Yes        | `"fdc3.timeRange"` |
| `startTime` | string *  | Yes        | `"2022-03-30T15:44:44+00:00"`       |
| `endTime`  | string *  | No        | `"2022-04-30T23:59:59+00:00"`            |

\* The time should be encoded in the [ISO 8601-1:2019](https://www.iso.org/standard/70907.html) format with a timezone indicator included

##### Example
A closed range:
```js
const timeRange = {
    type: "fdc3.timeRange",
    starttime: "2022-03-30T15:44:44+00:00",
    endtime: "2022-04-30T23:59:59+00:00"
}
```
An open range:
```js
const timeRange = {
    type: "fdc3.timeRange",
    starttime: "2022-03-30T15:44:44+00:00"
```

### Additional Information
The format of the strings representing times should refer to advice in the standard about formats, as proposed in: https://github.com/finos/FDC3/issues/645, with a note/link appearing in this context definition.

### LINKED ISSUE #651: Standardize a context type representing a Currency
## Enhancement Request

Create a context type that can be used to represent a Currency.

### Use Case:
A Currency context type is needed in order to compose other context types (such as an Order, Trade, Valuation or Position). However, an issue has also been raised to standardize recommended formats for 'primitive' fields, which might include ISO currency codes which may negate the need for this context as part of other context types. 
- #645 

A Currency context type may also be useful as a independent type, allowing it to be broadcast as part of workflows involving FOREX, liquidity pools or pricing. 

#### Currency
Represents and individual Currency.

##### Details

| Property      | Type    | Required | Example Value     |
|--------------|--------|-----------|---------------------|
| `type`           | string  | Yes          | `'fdc3.currency'`    |
| `name`         | string  | No          | `'US Dollar'`           |
| `ISOCODE` * | string  | No          | `'USD'`                   |

\* `ISOCODE` should conform to 3 character alphabetic codes standardized as part of ISO 4217. 

##### Example
```js
const currency = {
    type: 'fdc3.currency',
    name: 'US Dollar',
    ISOCODE: "USD"
}
```

### LINKED ISSUE #652: Standardize a context type representing a Valuation
## Enhancement Request
Create a context type representing the price and value of a holding. 

### Use Case:
A Valuation may be used as a key component of a Trade or Position context, perhaps also an Order. Although it may be used rarely as a standalone context (although use-cases may exist), it is a multi-field type that is likely to be needed in the definition of multiple other types, yielding an advantage to standardizing it.

#### Valuation
Represents the value of a holding in a specified currency.

##### Details

| Property    | Type    | Required | Example Value       |
|-------------|----------|----------|---------------------|
| `type`         | string    | Yes         | `'fdc3.valuation'`   |
| `value`       | number | Yes         | `'500.0'`                 |
| `price`        | number | No         | `'5.0'`                     |
| `CURRENCY_ISOCODE`  * | string  | Yes       | `'USD'`         |

\* CURRENCY_ISOCODE should conform to 3 character alphabetic codes standardized as part of ISO 4217.

##### Example
```js
const valuation = {
    type: 'fdc3.valuation',
    value: 500.0,
    price: 5.0,
    CURRENCY_ISOCODE: 'USD'
}
```

### Additional Information
An issue has also been raised to standardize recommended formats for 'primitive' fields, which might include ISO currency codes:
- #645 
 
If that issue is NOT accepted the CURRENCY_ISOCODE field should be replaced with a `currency` field that makes use of the Currency context type prosed in issue:
- #651

### LINKED ISSUE #654: Standardize a context type representing a Chart
## Enhancement Request

Create a context type representing details of a Chart, which may be used to request plotting of a particular chart or to otherwise share details of its composition.

Supercedes issue 
- #442 

### Use Case:
FDC3 already defines a `ViewChart` intent, which is typically provided with a basic Instrument context as input. This represents the bare minimum data needed to plot, for example, a pricing chart. However, there are many use-cases for more complex inputs to Charting application, such as:
- A list of instruments for comparison
- The time period to plot the chart over
- The style of chart (line, bar, mountain, candle etc.)
- Other settings such as indicators to calculate, or data representing drawings and annotations

Further, a charting application may wish to output a context representing what it is currently displaying so that it can be recorded by another application

#### ViewChart
(as defined in the standard already)

#### Chart
Represents the configuration of a Chart to be or that has been plotted.

##### Details

| Property    | Type    | Required | Example Value       |
|-------------|---------|----------|---------------------|
| `type`      | string  | Yes      | `'fdc3.chart'` |
| `instruments`      | InstrumentList  | Yes       | <pre>[<br>&emsp;&emsp;{<br>&emsp;&emsp;&emsp;&emsp;"type": "fdc3.instrument",<br>&emsp;&emsp;&emsp;&emsp;"id": {<br>&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;"ticker": "AAPL"<br>&emsp;&emsp;&emsp;&emsp;}<br>&emsp;&emsp;},<br>&emsp;&emsp;{<br>&emsp;&emsp;&emsp;&emsp;"type": "fdc3.instrument",<br>&emsp;&emsp;&emsp;&emsp;"id": {<br>&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;"ticker": "MSFT"<br>&emsp;&emsp;&emsp;&emsp;}<br>&emsp;&emsp;}<br>]</pre>       |
| `range` | TimeRange *  | No       | <pre>{<br>&emsp;&emsp;"type": "fdc3.timeRange",<br>&emsp;&emsp;"starttime": "2022-03-30T15:44:44+00:00",<br>&emsp;&emsp;"endtime": "2022-04-30T23:59:59+00:00"<br>}</pre>            |
| `style`    | string  | No       | `'line'` \| `'bar'` \| `'stacked-bar'` \| `'mountain'` \| `'candle'` \| `'pie'`        |
| `otherConfig`   | object ** | No |  `{ /* undefined object used to contain unstandardized config */}`  |

\* See issue:
- #646 
\** It is very common for charts to support other configurations, such as indicators, annotations etc., that a charting application may wish to take as input or produce as output, allowing the chart to fully configured and/or reproduced accurately. 

##### Example
```js
const chart = {
    type: "fdc3.chart",
    instruments: [
        {
            type: "fdc3.instrument",
            id: {
                ticker: "AAPL"
            }
        },
        {
            type: "fdc3.instrument",
            id: {
                ticker: "GOOG"
            }
        }
    ],
    range: {
        type: "fdc3.timeRange",
        starttime: "2020-09-01T08:00:00.000Z",
        endtime: "2020-10-31T08:00:00.000Z"
    },
    style: "line",
    otherConfig: {
      indicators: [
          {
              name: "ma",
              parameters: {
                  period: 14,
                  type: "ema"
              }
          },
          {
              name: "volume"
          }
      ]
    }
};

fdc3.raiseIntent("ViewChart", chart)
```

### Additional Information
The original version of this issue (#442) and its associated PR (#443) included the specification of an `Indicator` type, which are a key component of modern technical charts. As pushback was received on this element, it has been removed in favour of the `otherConfig` element, whose content is unstandardized, allowing applications that require this information to accurately reproduce a chart to still include it in this context.

### LINKED ISSUE #681: Intent naming conventions
## Enhancement Request
Provide guidance on the naming of intents.

- `View____` and `Start____` are the existing conventions
- Intents sometimes are meant to transfer focus and sometimes not, how do we represent/encode that?

Submitted on behalf of Riko Esteen (@rikoe )

### Use Case:
As the FDC3 standard matures, more use cases will evolve in turn identifying new intents that need to be created.  This issue has been raised to provide guidance on the naming of those intents.

New proposed conventions:
- `Create` / `Update` / `CreateOrUpdate` for e.g. orders, RFQs
- `Delete`
- `Get` (for two-way intents)
- `Share`

### PR DESCRIPTION
**Pull request by mistryvinay**
Consolidated PR for Context Data & Intents (CD&I) updates (each of which is raised as a separate PR into this branch for review by CD&I discussion group members, maintainers and editors).

resolves #574 
resolves #587
resolves #588 
resolves #589 
resolves #590 
resolves #600
resolves #624 
resolves #645
resolves #646
resolves #651
resolves #652
resolves #654 
resolves #681

### CODE REVIEW

**Comment by robmoffat** [comment]
Something going wrong with table rendering, for me at least (checked both Safari and Chrome, multiple pages).

e.g. https://deploy-preview-632--lambent-kulfi-cf51a7.netlify.app/docs/next/context/ref/instrumentlist

Screenshot 2022-05-24 at 11 52 13

**Comment by kriswest** [comment]
> Something going wrong with table rendering, for me at least (checked both Safari and Chrome, multiple pages).

Yes looks like it affects every table. Odd as its not doing it to me on a local build:
image

The generated HTML is identical but the CSS is different on the netlify build... it's been uglified and altered so that table is display `flex` rather than `block`

File: website/siteConfig.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ const siteConfig = {
   colors: {
     primaryColor: '#0086bf', /* This changes the top FINOS banner & all links - DO NOT CHANGE */
     secondaryColor: '#0033A0' /* This does net change anything - but keep as the project / program main color */
-  
+
```
Comment (kriswest) [inline website/siteConfig.js:50]:
```suggestion
```

File: src/context/ContextTypes.ts [hunk 2]
```diff
@@ -40,6 +40,14 @@ export interface ContactID {
   FDS_ID?: string;
 }
```
[the remark this answers is not in the document]
Comment (kriswest) [reply src/context/ContextTypes.ts:42]: Actual scratch that, the typegen won't work until this has been merged into main and deployed as the schemas are pulled from the website - hence nothing to do here as this is a release task.

File: src/intents/Intents.ts [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
   ViewContact = 'ViewContact',
   ViewQuote = 'ViewQuote',
   ViewNews = 'ViewNews',
   ViewInstrument = 'ViewInstrument',
   ViewAnalysis = 'ViewAnalysis',
   ViewHoldings = 'ViewHoldings',
+  ViewProfile = 'ViewProfile',
+  ViewResearch = 'ViewResearch',
+  ViewInteractions = 'ViewInteractions',
+  ViewOrders = 'ViewOrders'
```
Comment (kriswest) [inline src/intents/Intents.ts:15]: sort list:
```suggestion
export enum Intents {
  StartCall = 'StartCall',
  StartChat = 'StartChat',
  StartEmail = 'StartEmail',
  ViewAnalysis = 'ViewAnalysis',
  ViewChart = 'ViewChart',
  ViewContact = 'ViewContact',
  ViewHoldings = 'ViewHoldings',
  ViewInstrument = 'ViewInstrument',
  ViewInteractions = 'ViewInteractions',
  ViewNews = 'ViewNews',
  ViewOrders = 'ViewOrders'
  ViewProfile = 'ViewProfile',
  ViewQuote = 'ViewQuote',
  ViewResearch = 'ViewResearch',
```

**Review by kriswest — CHANGES_REQUESTED** [review]
Found a number of things to fix, including standardizing changelogs, sorting lists, incomplete list of standardized contexts etc.. Most have suggestions you can commit @mistryvinay 

I'll go fix the currency schema as I got that wrong.

**Review by mistryvinay — COMMENTED** [review]
Updated PR with suggested feedback and commits.

File: CHANGELOG.md [hunk 4]
```diff
@@ -33,8 +46,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 * Moved the Icon type definition into the Types documentation page for consistency with other types. ([#493](https://github.com/finos/FDC3/pull/493)
 * The `fdc3.joinChannel()`, `fdc3.getCurrentChannel()` and `fdc3.leaveCurrentChannel()` functions have been made optional for FDC3 API compliance, but are recommended through the application of the SHOULD keyword. ([#512](https://github.com/finos/FDC3/pull/512)) 
 * All DesktopAgent and Channel API functions are now async for consistency, changing the return type of the `broadcast`, `addIntentListener`, `addContextListener` and `getInfo` functions ([#516](https://github.com/finos/FDC3/pull/516))
+* Extended Intent Naming conventions and added hyperlinks for existing Intent spec definitions ([#701](https://github.com/finos/FDC3/pull/701))
+* Extended recommend field type conventions for contexts to include types for ids, times, dates, currency codes and country codes. The `fdc3.country` context type was updated to comply with the recommended field name for country codes (`COUNTRY_ISOALPHA2`). ([#704](https://github.com/finos/FDC3/pull/704))
```
Comment (kriswest) [inline CHANGELOG.md:50]:
```suggestion
* Extended recommended field type conventions for contexts to include types for ids, times, dates, currency codes and country codes. The `fdc3.country` context type was updated to comply with the recommended field name for country codes (`COUNTRY_ISOALPHA2`). ([#704](https://github.com/finos/FDC3/pull/704))
```

File: docs/context/spec.md [hunk 5]
```diff
@@ -60,28 +62,148 @@ The specification recognises that evolving context data definitions over time, a
...
+* Time in UTC: `"2022-03-30T15:44:44Z"`
+* Also time in UTC: `"2022-03-30T15:44:44+00:00"`
+* Same time in EDT: `"2022-03-30T11:44:44-04:00"`
+
+Times MAY be expressed with millisecond precision, e.g.:
+
+* `"2022-03-30T11:44:44.123-04:00"`
+* `"2022-03-30T11:44:44.123Z"`
+
+Parsing in Javascript:
```
Comment (greyseer256) [inline docs/context/spec.md:106]:
```suggestion
Parsing in JavaScript:
```

File: docs/context/spec.md [hunk 6]
```diff
@@ -60,28 +62,148 @@ The specification recognises that evolving context data definitions over time, a
...
+let aDate = new Date("2022-03-30T11:44:44.123-04:00")
+```
+
+### Dates
+
+Fields representing a point in time SHOULD be string encoded using the `YYYY-MM-DD` date format from [ISO 8601-1:2019](https://www.iso.org/standard/70907.html).
+
+E.g. `"2022-03-30"`
+
+Parsing in Javascript:
```
Comment (greyseer256) [inline docs/context/spec.md:118]:
```suggestion
Parsing in JavaScript:
```

File: docs/intents/ref/StartChat.md [hunk 7]
```diff
@@ -33,13 +34,42 @@ const contact = {
...
+                email: 'john@mail.com'
+            },
+        }]
+    },
+    options: {
+        groupRecipients: true, // one chat with both contacts
+        public: false, // private chat room
+        allowHistoryBrowsing: true,
+        allowMessageCopy: true,
+        allowAddUser: false, // John won't be auhtorized to add other users to the chat
```
Comment (greyseer256) [inline docs/intents/ref/StartChat.md:57]:
```suggestion
        allowAddUser: false, // John won't be authorized to add other users to the chat
```

**Review by greyseer256 — APPROVED** [review]
Minor comments, looks good.

**Pull request merged**